packages/component-library/src/components/plmg-text-input/plmg-text-input.tsx

```diff
@@ -195,17 +408,50 @@ export class TextInput {
           {this.showLabel && this.required && <span class={'required'}>*</span>}
         </label>
         <div class={'plmg-text-input-field-wrapper'} tabIndex={0}>
-          <input
-            disabled={this.disabled}
-            class={inputClasses}
-            id={this.labelToId()}
-            name={this.label}
-            required={this.required}
-            type={'text'}
-            readonly={this.readOnly}
-            value={this.internalValue}
-            onInput={(ev) => this.handleInputChange(ev)}
-          />
+          {this.type === 'number' ? (
+            <input
+              autoComplete={this.autoComplete}
+              class={inputClasses}
+              disabled={this.disabled}
+              id={this.labelToId()}
+              max={this.max}
```

Comment:
I'm not sure you need to separate the 2 cases. Have you try providing all the attributes regardless of the type?
I'm pretty sure a input of type number will ignore `maxLength` for example.
And an input of type password will ignore `max` and `min`.

Reply:
and it leads to duplication problem, for example you forgot to duplicate the `width` logic.

Reply:
Yes. Works fine. Tests show that indeed irrelevant props are ignored. I was using the 'size' attribute to set the width in preference to inline styling, but that has no effect on the number.